### DISCUSSION TITLE
Feature : Add Auto Tiling templates

### DISCUSSION THREAD

**Discussion by nikhilCad** [Orama-Interactive/Pixelorama#564] [Ideas]
Stretch goal:
Pixelorama is a tool for game developers. Many game developers are using auto tiles for their levels to maintain their sanity.
Here is a 47 tile setup:
![](http://www.ekalia.com/gm/autotiles_helper/doc/img/expanded.png)
And a 16 tile setup:
![image](https://user-images.githubusercontent.com/47214394/71317480-bd29e800-24a7-11ea-83d8-b9cc8fdd5b2c.png)
They are fairly easy to use(in Godot and GMS2 at least). But the real challenge is in making them.

Pixelorama can have a simple option to enable a 'reference background/template':
47 tiles:
![image](https://user-images.githubusercontent.com/47214394/71317411-7982ae80-24a6-11ea-897c-644785a87afd.png)
16 tiles(Please dont mind the numbers at corners, this is the only image I could find):
![](https://www.codeproject.com/KB/game/Autotiles_Algorithm/AutoTileSequence.png)

There should be an option to select the size of individual tile(16x16, 32x32 etc.)
Again, a stretch goal. Would be nice to have, and very useful too.

**Comment by OverloadedOrama** [comment]
This sounds very interesting, but probably not something that will happen soon! I'd love to experiment with this in the future though!

**Comment by Calinou** [comment]
@OverloadedOrama Would it be possible to include a few `.pxo` files in a `Templates/` folder included in the Pixelorama distribution? This way, we could have an "open template" menu that opens the file browser directly in that folder.

Alternatively, we can include a few PNGs in there and make "Open template" import the PNG as an image. This way, it can be imported into an existing image.

**Comment by Erevoid** [comment]
Actually, a tilemap editor could make a great future addition and a very useful tool
